server/app/services/program/ProgramServiceImpl.java

```diff
@@ -210,7 +216,17 @@ public ErrorAnd<ProgramDefinition, CiviFormError> updateLocalization(
   }
 
   private boolean hasProgramNameCollision(String programName) {
-    return getActiveAndDraftPrograms().getProgramNames().contains(programName);
+    ImmutableSet<String> programNames = getActiveAndDraftPrograms().getProgramNames();
+
+    if (programNames.contains(programName)) {
+      return true;
+    }
+
+    Slugify slugifier = new Slugify();
```

Comment:
nit, optional: Consider adding a comment with example to explain what this is to people like me (who just learned what slugging is :)).

e.g. Something like

Compare slugified version of programName, which is what is used in the url. e.g. "my program" and "my  program" will both be slugified to "my-program".

Reply:
done